conf/fe.conf

```diff
@@ -24,7 +24,7 @@
 LOG_DIR = ${STARROCKS_HOME}/log
 
 DATE = "$(date +%Y%m%d-%H%M%S)"
-JAVA_OPTS="-Dlog4j2.formatMsgNoLookups=true -Xmx8192m -XX:+UseG1GC -Xlog:gc*:${LOG_DIR}/fe.gc.log.$DATE:time -XX:ErrorFile=${LOG_DIR}/hs_err_pid%p.log -Djava.security.policy=${STARROCKS_HOME}/conf/udf_security.policy"
+JAVA_OPTS="-Dlog4j2.formatMsgNoLookups=true -Xmx8192m -XX:+UseG1GC -Xlog:gc*:${LOG_DIR}/fe.gc.log.$DATE:time -XX:ErrorFile=${LOG_DIR}/hs_err_pid%p.log -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${LOG_DIR} -XX:+ExitOnOutOfMemoryError -Djava.security.policy=${STARROCKS_HOME}/conf/udf_security.policy"
```

Comment:
what's the expected size of the heapDump? what if it crashes repeatedly that will fill up the disk quickly?

Reply:
It is ok to turn on the heap dump as necessary for debugging purpose. However the default value should be production-oriented. I am suggesting manual set the JAVA_OPTS as necessary but leave the default value in codebase unchanged.

Reply:
> It is ok to turn on the heap dump as necessary for debugging purpose. However the default value should be production-oriented. I am suggesting manual set the JAVA_OPTS as necessary but leave the default value in codebase unchanged.

thanks @kevincai . I think this option is useful even in a production-oriented, which is sometimes not easy to reproduce. In addition, it will only work when OOM occurs and will not affect normal production environment use. Therefore, most JAVA-based  components will use it as the default option.

Reply:
So, it is certainly helpful with troubleshoot.

Here what I look at this change.

1), is it a blocker if without the code change? will it prevent the end user to turn on this ability without code change?
Apparently, the answer is "NO".

2), is it a good practice to turn it on by default in production?
Should address the concern of disk space usage if turned on by default. 

To me, the end user can turn it on as necessary when facing production issue. But mostly there is no need for this heap dump, especially when the JVM mem size up to hundreds of GB, the dump itself will take time. In production, the top priority is to get the process up again as quick as possible in most cases.

Reply:
> So, it is certainly helpful with troubleshoot.
> 
> Here what I look at this change.
> 
> 1), is it a blocker if without the code change? will it prevent the end user to turn on this ability without code change? Apparently, the answer is "NO".
> 
> 2), is it a good practice to turn it on by default in production? Should address the concern of disk space usage if turned on by default.
> 
> To me, the end user can turn it on as necessary when facing production issue. But mostly there is no need for this heap dump, especially when the JVM mem size up to hundreds of GB, the dump itself will take time. In production, the top priority is to get the process up again as quick as possible in most cases.

Thanks for the explanation